### ISSUE TITLE
Sample Code Chapter 4

### ISSUE THREAD

**Issue by sthzg** [habuma/spring-in-action-4-samples#8]
Is there a reason why sample code from chapter 4 is not available?

**Comment by Sanjay-F** [comment]
+1
wanna know why too.

**Comment by anthony-cros** [comment]
+1

**Comment by ghost** [comment]
+1

**Comment by arcioneo** [comment]
Yeah, I don't know how to finish the example when you use @DeclareParents

**Comment by AshkinLi** [comment]
+1
Has some example about @DeclareParents ?

**Comment by ZaynChen** [comment]
`
public class DefaultEncoreable implements Encoreable {

@Override
public void performEncore() {
System.out.println("encore");
}
}`

`
@Aspect
public class EncoreableIntroducer {

@DeclareParents(value = "chapter_4_concert.domain.Performance+", defaultImpl = DefaultEncoreable.class)
public static Encoreable encoreable;
}
`
using:
`
@Autowired
private Performance performance;
`
`
Encoreable encore = (Encoreable) performance;
encore.performEncore();
assertEquals("encore\n", systemOutRule.getLog());
`
These are my idea about using @DeclareParents.

**Comment by yawak** [comment]
+1

Can we have a sample code for chapter 4 that includes test?

**Comment by aCodeRancher** [comment]
Is chapter 4 example code missing ?
